python/ray/air/config.py

```diff
@@ -185,14 +193,21 @@ def _resources_per_worker_not_none(self):
                 # Note that we don't request any CPUs, which avoids possible
                 # scheduling contention. Generally nodes have many more CPUs than
                 # GPUs, so not requesting a CPU does not lead to oversubscription.
-                return {"GPU": 1}
+                resources_per_worker = {"GPU": 1}
             else:
-                return {"CPU": 1}
-        resources_per_worker = {
-            k: v for k, v in self.resources_per_worker.items() if v != 0
-        }
+                resources_per_worker = {"CPU": 1}
+        else:
+            resources_per_worker = {
+                k: v for k, v in self.resources_per_worker.items() if v != 0
+            }
+
         if self.use_gpu:
             resources_per_worker.setdefault("GPU", 1)
+
+        if self.accelerator_type:
+            resources_per_worker[
+                f"{RESOURCE_CONSTRAINT_PREFIX}{self.accelerator_type}"
+            ] = 0.001
```

Comment:
Should we use `setdefault` similar to GPU in the line above? In case the user manually sets `resources_per_worker`. Or validate that it's not set in the resources dict if we want this to be the only interface where the user can specify.